pkg/component/component.go

```diff
@@ -402,6 +339,215 @@ func getS2IPaths(podEnvs []corev1.EnvVar) []string {
+		componentSettings.Application == nil ||
+		componentSettings.Project == nil ||
+		componentSettings.SourceType == nil ||
+		componentSettings.SourceLocation == nil ||
+		componentSettings.Type == nil {
+		return fmt.Errorf("component name, application, project, source type, source location and component type are mandatory parameters for component creation")
+	}
+
+	_, componentType, _, componentVersion := util.ParseComponentImageName(*componentSettings.Type)
+	// Check to see if the catalog type actually exists
+	exists, err := catalog.Exists(client, componentType)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to create component of type %s", componentType)
+	}
+	if !exists {
+		log.Info("Run 'odo catalog list components' for a list of supported component types")
+		return fmt.Errorf("Failed to find component of type %s", componentType)
+	}
+
+	// Check to see if that particular version exists
+	versionExists, err := catalog.VersionExists(client, componentType, componentVersion)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to create component of type %s of version %s", componentType, componentVersion)
+	}
+	if !versionExists {
+		log.Info("Run 'odo catalog list components' to see a list of supported component type versions")
+		return fmt.Errorf("Invalid component version %s:%s", componentType, componentVersion)
+	}
+
+	// Validate component name
+	err = validation.ValidateName(*componentSettings.Name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create component of name %s", *componentSettings.Name)
+	}
+
+	if isCmpExistsCheck {
+		exists, err = Exists(client, *componentSettings.Name, *componentSettings.Application)
+		if err != nil {
+			return errors.Wrapf(err, "failed to check if component of name %s exists in application %s", *componentSettings.Name, *componentSettings.Application)
+		}
+		if exists {
+			return fmt.Errorf("component with name %s already exists in application %s", *componentSettings.Name, *componentSettings.Application)
+		}
+	}
+
+	if *componentSettings.SourceType == config.LOCAL {
+		srcLocInfo, err := os.Stat(*(componentSettings.SourceLocation))
+		if err != nil {
+			return errors.Wrapf(err, "failed to create component with settings %+v", componentSettings)
+		}
+		if !srcLocInfo.IsDir() {
+			return fmt.Errorf("source path for component created for local source needs to be a directory")
+		}
+	}
+
+	return
+}
+
+// ApplyConfig applies the component config onto component dc
+// Parameters:
+//	client: occlient instance
+//	appName: Name of application of which the component is a part
+//	componentName: Name of the component which is being patched with config
+//	componentConfig: Component configuration
+// Returns:
+//	err: Errors if any else nil
+func ApplyConfig(client *occlient.Client, componentConfig config.LocalConfigInfo, context string, stdout io.Writer) (err error) {
+
+	cmpName := componentConfig.GetName()
+	cmpSrcLoc := componentConfig.GetSourceLocation()
+	if err != nil {
```

Comment:
I guess this check is unnecessary

Reply:
yeah you are right
It was probably just a residue from a previous implementation
Removed now